components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/streaming/SubscriptionHelper.java

```diff
@@ -165,6 +165,7 @@ public void onMessage(ClientSessionChannel channel, Message message) {
                             LOG.warn("Connect failure: {}", message);
                             connectError = (String) message.get(ERROR_FIELD);
                             connectException = getFailure(message);
+                            client.disconnect();
```

Comment:
Should this be in a try .. catch and ignore (I wonder if disconnect can throw an exception)